### PR TITLE
feat: enabled s2i for python

### PR DESCRIPTION
**Pull request by matejvasek**
# Changes

- :gift: Enabled s2i build for Python functions.

```release-note
feat: enabled s2i for python functions
```

### CODE REVIEW

File: pkg/builders/s2i/builder_test.go [hunk 1]
```diff
@@ -63,11 +63,6 @@ func Test_BuildImages(t *testing.T) {
 			function: fn.Function{Runtime: "go"},
 			wantErr:  true,
 		},
-		{
-			name:     "Without builder - unsupported runtime - python",
-			function: fn.Function{Runtime: "python"},
-			wantErr:  true,
-		},
```
Comment (lance) [inline pkg/builders/s2i/builder_test.go:70]: Could you change this instead to a test for python that is supported?

**Review by zroubalik — COMMENTED** [review]
Looking good! Could you please also enable e2e tests for this? https://github.com/knative/func/blob/main/test/oncluster/scenario_runtime_test.go#L21

**Comment by matejvasek** [comment]
@lance @zroubalik @jrangelramos PTAL

**Comment by matejvasek** [comment]
@lance could you please override `codecov/project` ? The coverage upload is flaking.

**Review by lance — COMMENTED** [review]
/lgtm
Thanks @matejvasek!

**Review by zroubalik — APPROVED** [review]
/lgtm

awesome!

**Pull request merged**